### ISSUE TITLE
[3.0.0] java.lang.RuntimeException: Could not invoke RNAdapty.handle

### ISSUE THREAD

**Issue by efstathiosntonas** [adaptyteam/AdaptySDK-React-Native#131]
### Description

after upgrading from v2 to v3 on Android it throws this error on Pixel 7Pro Android 14 emulator:

```logcat
Exception in native call (Ask Gemini)
      java.lang.RuntimeException: Could not invoke RNAdapty.handle
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:381)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146)
       at com.facebook.jni.NativeRunnable.run(Native Method)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
       at android.os.Looper.loopOnce(Looper.java:205)
       at android.os.Looper.loop(Looper.java:294)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235)
       at java.lang.Thread.run(Thread.java:1012)
      Caused by: java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Method.invoke(Native Method)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
      Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/adapty/ui/AdaptyUI$LocalizedViewConfiguration$Screen$Default;
       at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16)
       at com.google.gson.internal.Excluder.excludeField(Excluder.java:184)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161)
       at com.google.gson.Gson.getAdapter(Gson.java:628)
       at com.google.gson.Gson.toJson(Gson.java:928)
       at com.google.gson.Gson.toJson(Gson.java:899)
       at com.google.gson.Gson.toJson(Gson.java:848)
       at com.google.gson.Gson.toJson(Gson.java:825)
       at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100)
       at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35)
       at com.adapty.react.AdaptyContext.reject(AdaptyContext.kt:53)
       at com.adapty.react.AdaptyContext.bridgeError(AdaptyContext.kt:84)
       at com.adapty.react.AdaptyCallHandler.handle(AdaptyCallHandler.kt:49)
       at com.adapty.react.AdaptyReactModule.handle(AdaptyReactModule.kt:91)
       at java.lang.reflect.Method.invoke(Native Method) 
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372) 
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
      Caused by: java.lang.ClassNotFoundException: Didn't find class "com.adapty.ui.AdaptyUI$LocalizedViewConfiguration$Screen$Default" on path: DexPathList[[zip file "/data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/base.apk"],nativeLibraryDirectories=[/data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/lib/arm64, /data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
       at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:637)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:573)
       at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16) 
       at com.google.gson.internal.Excluder.excludeField(Excluder.java:184) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161) 
       at com.google.gson.Gson.getAdapter(Gson.java:628) 
       at com.google.gson.Gson.toJson(Gson.java:928) 
       at com.google.gson.Gson.toJson(Gson.java:899) 
       at com.google.gson.Gson.toJson(Gson.java:848) 
       at com.google.gson.Gson.toJson(Gson.java:825) 
       at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100) 
       at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35) 
       at com.adapty.react.AdaptyContext.reject(AdaptyContext.kt:53) 
       at com.adapty.react.AdaptyContext.bridgeError(AdaptyContext.kt:84) 
       at com.adapty.react.AdaptyCallHandler.handle(AdaptyCallHandler.kt:49) 
       at com.adapty.react.AdaptyReactModule.handle(AdaptyReactModule.kt:91) 
       at java.lang.reflect.Method.invoke(Native Method) 
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372) 
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
2024-10-23 11:43:57.643 21646-21755 unknown:ReactNative     com.APP_NAME                             E  Exception in native call (Ask Gemini)
      java.lang.RuntimeException: Could not invoke RNAdapty.handle
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:381)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146)
       at com.facebook.jni.NativeRunnable.run(Native Method)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
       at android.os.Looper.loopOnce(Looper.java:205)
       at android.os.Looper.loop(Looper.java:294)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235)
       at java.lang.Thread.run(Thread.java:1012)
      Caused by: java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Method.invoke(Native Method)
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
      Caused by: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/adapty/ui/AdaptyUI$LocalizedViewConfiguration$Screen$Default;
       at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16)
       at com.google.gson.internal.Excluder.excludeField(Excluder.java:184)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347)
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161)
       at com.google.gson.Gson.getAdapter(Gson.java:628)
       at com.google.gson.Gson.toJson(Gson.java:928)
       at com.google.gson.Gson.toJson(Gson.java:899)
       at com.google.gson.Gson.toJson(Gson.java:848)
       at com.google.gson.Gson.toJson(Gson.java:825)
       at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100)
       at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35)
       at com.adapty.react.AdaptyContext.reject(AdaptyContext.kt:53)
       at com.adapty.react.AdaptyContext.bridgeError(AdaptyContext.kt:84)
       at com.adapty.react.AdaptyCallHandler.handle(AdaptyCallHandler.kt:49)
       at com.adapty.react.AdaptyReactModule.handle(AdaptyReactModule.kt:91)
       at java.lang.reflect.Method.invoke(Native Method) 
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372) 
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
      Caused by: java.lang.ClassNotFoundException: Didn't find class "com.adapty.ui.AdaptyUI$LocalizedViewConfiguration$Screen$Default" on path: DexPathList[[zip file "/data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/base.apk"],nativeLibraryDirectories=[/data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/lib/arm64, /data/app/~~D-yzIxmOUR93cv_ZG4ehPA==/com.APP_NAME-v7HkH07urKd29-TJ9FVnBA==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64]]
       at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:637)
       at java.lang.ClassLoader.loadClass(ClassLoader.java:573)
       at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16) 
       at com.google.gson.internal.Excluder.excludeField(Excluder.java:184) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347) 
       at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161) 
       at com.google.gson.Gson.getAdapter(Gson.java:628) 
       at com.google.gson.Gson.toJson(Gson.java:928) 
       at com.google.gson.Gson.toJson(Gson.java:899) 
       at com.google.gson.Gson.toJson(Gson.java:848) 
       at com.google.gson.Gson.toJson(Gson.java:825) 
       at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100) 
       at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35) 
       at com.adapty.react.AdaptyContext.reject(AdaptyContext.kt:53) 
       at com.adapty.react.AdaptyContext.bridgeError(AdaptyContext.kt:84) 
       at com.adapty.react.AdaptyCallHandler.handle(AdaptyCallHandler.kt:49) 
       at com.adapty.react.AdaptyReactModule.handle(AdaptyReactModule.kt:91) 
       at java.lang.reflect.Method.invoke(Native Method) 
       at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372) 
       at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:146) 
       at com.facebook.jni.NativeRunnable.run(Native Method) 
       at android.os.Handler.handleCallback(Handler.java:958) 
       at android.os.Handler.dispatchMessage(Handler.java:99) 
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27) 
       at android.os.Looper.loopOnce(Looper.java:205) 
       at android.os.Looper.loop(Looper.java:294) 
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:235) 
       at java.lang.Thread.run(Thread.java:1012) 
2024-10-23 11:43:57.658 21646-21755 Bugsnag                 com.APP_NAME                             D  Received MessageEvent: ContextUpdate
2024-10-23 11:43:57.666 21646-21845 chromium                com.APP_NAME                             E  [ERROR:simple_file_enumerator.cc(21)] opendir /data/user/0/com.APP_NAME/cache/WebView/Default/HTTP Cache/Code Cache/js: No such file or directory (2)
2024-10-23 11:43:57.666 21646-21845 chromium                com.APP_NAME                             E  [ERROR:simple_index_file.cc(620)] Could not reconstruct index from disk
2024-10-23 11:43:57.668 21646-21867 chromium                com.APP_NAME                             E  [ERROR:simple_file_enumerator.cc(21)] opendir /data/user/0/com.APP_NAME/cache/WebView/Default/HTTP Cache/Code Cache/wasm: No such file or directory (2)
2024-10-23 11:43:57.668 21646-21867 chromium                com.APP_NAME                             E  [ERROR:simple_index_file.cc(620)] Could not reconstruct index from disk
```

### Version

v3.0.0

### What platforms are you seeing the problem on?

Android

### System info

```yml
System:
  OS: macOS 15.0.1
  CPU: (20) arm64 Apple M1 Ultra
  Memory: 79.56 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.17.0
    path: ~/.nvm/versions/node/v20.17.0/bin/node
  Yarn:
    version: 4.4.1
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.9.0
    path: ~/.nvm/versions/node/v20.17.0/bin/npm
  Watchman:
    version: 2024.10.21.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /Users/stathis/.gem/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK:
    API Levels:
      - "31"
      - "33"
      - "34"
      - "35"
    Build Tools:
      - 30.0.3
      - 33.0.0
      - 33.0.1
      - 33.0.2
      - 34.0.0
    System Images:
      - android-25 | Google APIs ARM 64 v8a
      - android-27 | Google APIs ARM 64 v8a
      - android-28 | Google APIs ARM 64 v8a
      - android-29 | Google APIs ARM 64 v8a
      - android-30 | Google APIs ARM 64 v8a
      - android-31 | Google APIs ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12483815
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.13
    path: /usr/bin/javac
  Ruby:
    version: 3.3.5
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.4
    wanted: 0.75.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

**Comment by efstathiosntonas** [comment]
on a real device, Samsung S22 with Android 14 I get this stacktrace: 

```
FATAL EXCEPTION: main (Ask Gemini)
    Process: com.APP_NAME, PID: 30900
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/adapty/ui/AdaptyUI$LocalizedViewConfiguration$Screen$Default;
     at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16)
     at com.google.gson.internal.Excluder.excludeField(Excluder.java:184)
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81)
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347)
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161)
     at com.google.gson.Gson.getAdapter(Gson.java:628)
     at com.google.gson.Gson.toJson(Gson.java:928)
     at com.google.gson.Gson.toJson(Gson.java:899)
     at com.google.gson.Gson.toJson(Gson.java:848)
     at com.google.gson.Gson.toJson(Gson.java:825)
     at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100)
     at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35)
     at com.adapty.react.AdaptyReactModule$callHandler$1.invoke$lambda$0(AdaptyReactModule.kt:106)
     at com.adapty.react.AdaptyReactModule$callHandler$1.$r8$lambda$irmzzz6CwmLJA34D_JLUly-OMHw(Unknown Source:0)
     at com.adapty.react.AdaptyReactModule$callHandler$1$$ExternalSyntheticLambda0.onProfileReceived(D8$$SyntheticClass:0)
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2.invokeSuspend$lambda$0(AdaptyInternal.kt:50)
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2.$r8$lambda$60eHZlqC9s5ecb-y5aauxu_TtJE(Unknown Source:0)
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
     at android.os.Handler.handleCallback(Handler.java:958)
     at android.os.Handler.dispatchMessage(Handler.java:99)
     at android.os.Looper.loopOnce(Looper.java:230)
     at android.os.Looper.loop(Looper.java:319)
     at android.app.ActivityThread.main(ActivityThread.java:8919)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "com.adapty.ui.AdaptyUI$LocalizedViewConfiguration$Screen$Default" on path: DexPathList[[zip file "/data/app/~~kjL-2w7yzz9LsOcsOjEPiA==/com.APP_NAME-eoZlZqKMg6COazR6zTY6RQ==/base.apk"],nativeLibraryDirectories=[/data/app/~~kjL-2w7yzz9LsOcsOjEPiA==/com.APP_NAME-eoZlZqKMg6COazR6zTY6RQ==/lib/arm64, /data/app/~~kjL-2w7yzz9LsOcsOjEPiA==/com.APP_NAME-eoZlZqKMg6COazR6zTY6RQ==/base.apk!/lib/arm64-v8a, /system/lib64]]
     at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
     at java.lang.ClassLoader.loadClass(ClassLoader.java:637)
     at java.lang.ClassLoader.loadClass(ClassLoader.java:573)
     at com.adapty.internal.crossplatform.SerializationExclusionStrategy.shouldSkipField(SerializationExclusionStrategy.kt:16) 
     at com.google.gson.internal.Excluder.excludeField(Excluder.java:184) 
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.includeField(ReflectiveTypeAdapterFactory.java:81) 
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:347) 
     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:161) 
     at com.google.gson.Gson.getAdapter(Gson.java:628) 
     at com.google.gson.Gson.toJson(Gson.java:928) 
     at com.google.gson.Gson.toJson(Gson.java:899) 
     at com.google.gson.Gson.toJson(Gson.java:848) 
     at com.google.gson.Gson.toJson(Gson.java:825) 
     at com.adapty.internal.crossplatform.SerializationHelper.toJson(SerializationHelper.kt:100) 
     at com.adapty.internal.crossplatform.CrossplatformHelper.toJson(CrossplatformHelper.kt:35) 
     at com.adapty.react.AdaptyReactModule$callHandler$1.invoke$lambda$0(AdaptyReactModule.kt:106) 
     at com.adapty.react.AdaptyReactModule$callHandler$1.$r8$lambda$irmzzz6CwmLJA34D_JLUly-OMHw(Unknown Source:0) 
     at com.adapty.react.AdaptyReactModule$callHandler$1$$ExternalSyntheticLambda0.onProfileReceived(D8$$SyntheticClass:0) 
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2.invokeSuspend$lambda$0(AdaptyInternal.kt:50) 
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2.$r8$lambda$60eHZlqC9s5ecb-y5aauxu_TtJE(Unknown Source:0) 
     at com.adapty.internal.AdaptyInternal$onProfileUpdatedListener$1$2$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0) 
     at android.os.Handler.handleCallback(Handler.java:958) 
     at android.os.Handler.dispatchMessage(Handler.java:99) 
     at android.os.Looper.loopOnce(Looper.java:230) 
     at android.os.Looper.loop(Looper.java:319) 
     at android.app.ActivityThread.main(ActivityThread.java:8919) 
     at java.lang.reflect.Method.invoke(Native Method) 
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578) 
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 
```

**Comment by efstathiosntonas** [comment]
It seems that  `@adapty/react-native-ui` is a required dep on v3.

**Comment by vladd-g** [comment]
@efstathiosntonas thank you! Could you please try v3.0.1? without including `@adapty/react-native-ui`

**Comment by efstathiosntonas** [comment]
hi @vladd-g, I did `yarn remove @adapty/react-native-ui && yarn add react-native-adapty@3.0.1` and rebuilt on Android, everything is fine now,  I've had to clean build though first.

**Comment by efstathiosntonas** [comment]
closing as this is fixed on v3.0.1
